From b1849a18cf6daa98f0f10860dfd66da2a2f591fb Mon Sep 17 00:00:00 2001
From: Lily Foster <lily@lily.flowers>
Date: Tue, 21 Feb 2023 20:25:36 -0500
Subject: [PATCH] xdg-utils: patch xdg-open to correctly open files when using
 portal

See upstream MR at https://gitlab.freedesktop.org/xdg/xdg-utils/-/merge_requests/65 which adds support for the org.freedesktop.portal.OpenURI.OpenFile method.
---
 pkgs/tools/X11/xdg-utils/default.nix | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/pkgs/tools/X11/xdg-utils/default.nix b/pkgs/tools/X11/xdg-utils/default.nix
index 53a43dde3030b..c3b1dd546fb72 100644
--- a/pkgs/tools/X11/xdg-utils/default.nix
+++ b/pkgs/tools/X11/xdg-utils/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitLab, fetchFromGitHub
+{ lib, stdenv, fetchFromGitLab, fetchFromGitHub, fetchpatch
 , file, libxslt, docbook_xml_dtd_412, docbook_xsl, xmlto
 , w3m, gnugrep, gnused, coreutils, xset, perlPackages
 , mimiSupport ? false, gawk
@@ -36,6 +36,13 @@ stdenv.mkDerivation rec {
     # Allow forcing the use of XDG portals using NIXOS_XDG_OPEN_USE_PORTAL environment variable.
     # Upstream PR: https://github.com/freedesktop/xdg-utils/pull/12
     ./allow-forcing-portal-use.patch
+    # Allow opening files when using portal with xdg-open.
+    # Upstream PR: https://gitlab.freedesktop.org/xdg/xdg-utils/-/merge_requests/65
+    (fetchpatch {
+      name = "support-openfile-with-portal.patch";
+      url = "https://gitlab.freedesktop.org/xdg/xdg-utils/-/commit/5cd8c38f58d9db03240f4bc67267fe3853b66ec7.diff";
+      hash = "sha256-snkhxwGF9hpqEh5NGG8xixTi/ydAk5apXRtgYrVgNY8=";
+    })
   ];
 
   # just needed when built from git
